projects/packages/my-jetpack/_inc/hooks/use-analytics/index.ts

```diff
@@ -40,16 +40,19 @@ const useAnalytics = () => {
 	 * @param {string} event      - event name
 	 * @param {object} properties - event propeties
 	 */
-	const recordEvent = useCallback< TracksRecordEvent >( ( event, properties ) => {
-		jetpackAnalytics.tracks.recordEvent( event, {
-			...properties,
-			version: myJetpackVersion,
-			is_site_connected: isSiteConnected,
-			is_user_connected: isUserConnected,
-			referring_plugins: connectedPluginsSlugs,
-		} );
+	const recordEvent = useCallback< TracksRecordEvent >(
+		( event, properties ) => {
+			jetpackAnalytics.tracks.recordEvent( event, {
+				...properties,
+				version: myJetpackVersion,
+				is_site_connected: isSiteConnected,
+				is_user_connected: isUserConnected,
+				referring_plugins: connectedPluginsSlugs,
+			} );
+		},
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [] );
+		[ isSiteConnected ]
```

Comment:
When tracks was being sideloaded after a site connection, the `is_site_connected` value was false incorrectly